uni/lib/model/entities/calendar_event.dart

```diff
@@ -1,12 +1,35 @@
+import 'package:intl/intl.dart';
+
 /// An event in the school calendar
 class CalendarEvent {
   /// Creates an instance of the class [CalendarEvent]
-  CalendarEvent(this.name, this.date);
+  CalendarEvent(this.name, this.date, {this.closeDate}) {
+    name = name;
+    date = date;
+    closeDate = getDateTime();
+  }
   String name;
   String date;
+  DateTime? closeDate;
 
   /// Converts the event into a map
   Map<String, dynamic> toMap() {
     return {'name': name, 'date': date};
   }
+
+  DateTime? getDateTime() {
```

Comment:
[This](https://flutterbyexample.com/lesson/getters-and-setters) is more idiomatic.